Documentation: Adding Virtual Environment Instructions to README.md

npvq opened the pull request:
Virtual Environment Instructions added for Linux and OS X only.

New contributors should use a Virtual Environment to make sure their python installation has the same packages as the requirements, and should they use a new package, they could update `requirements.txt` via `pip freeze`. The `README.md` installation documentation was updated for Linux and OS X with instructions to set up a Virtual Environment via `venv` for Python 3.

npvq committed:
Add Virtual Environment Instructions to README.md

andria-dev reviewed (changes requested):
Hi @npvq, I feel like the additions to the README would be better off in their own section after the section for regular usage. For example, something to the effect of the following Markdown would be good. Also, the path `AbBOT` needs to be updated to `AbBOT-python` in all of the examples.

## Using a Virtual Environment

Why you should use a virtual environment as a developer.

### Setup

#### Linux and macOS

```bash
# stuff with python
```

#### Windows

```cmd
:: stuff with cmd.exe
```

npvq committed:
Moved Virtual Environment Documentation

Virtual Environment Documentation has been moved after regular usage as requested, and relevant documentation for Windows has been added. Instances of `AbBOT` were updated to `AbBOT-python`. The new sections have been manually added to TOC, and the Virtual Environment section has been hyperlinked before the installation of the packages in `requirements.txt` in every relevant operating system.

npvq committed:
Fixed "macOS" Typo

Replaced instances of "MacOS" with "macOS."

npvq commented:
Thank you for the feedback. I have made the requested changes. Note that the "Linux and macOS" and "Windows" sections were named `Setup for Linux and macOS` and `Setup for Windows` respectively to avoid conflict with the `Windows` section in the TOC. Please let me know if I need to make any other changes.

LakesideMiners reviewed (approved):
all seems good here, merging
